ledger-api/grpc-definitions/com/daml/ledger/api/v1/admin/object_meta.proto

```diff
@@ -0,0 +1,62 @@
+message ObjectMeta {
+
+  // An opaque value populated by a participant server which represents the internal version of the resource
+  // this ObjectMeta message is attached to. The participant server will change it to a unique value each time the corresponding resource is updated.
+  // You must not rely on the format of resource version. The participant server might change it without notice.
+  //
+  // You can obtain the newest resource version value by issuing a read request.
+  // You may use it for concurrent change detection by passing it back unmodified in an update request.
+  // The participant server will compare the passed value with the value maintained by the system to determine
+  // if any other updates took place since you read the resource version.
+  // Upon a successful update you are guaranteed that no other update took place during your read-modify-write sequence.
+  // However, if another update took place during your read-modify-write sequence then your update will fail with an appropriate error.
+  // You may also omit it from an update request which will result in no concurrent change detection being applied.
+  //
+  // When creating a new instance of a resource you must leave it empty. Its value will be populated by the participant server
+  // upon successful resource creation.
+  //
+  // Optional
+  string resource_version = 6;
+
+  // A set of modifiable key-value pairs that can be used to represent arbitrary, client-specific metadata.
+  //
+  // Constraints:
+  // 1. The total size over all keys and values cannot exceed 256kb in UTF-8 encoding.
+  // 2. Keys are composed of an optional prefix segment and a required name segment such that:
+  //    - key prefix, when present, must be a valid DNS subdomain with at most 253 characters, followed by a '/' (forward slash) character,
+  //    - name segment must have at most 63 characters that are either alphanumeric ([a-z0-9A-Z]), or a '.' (dot), '-' (dash) or '_' (underscore);
+  //      and it must start and end with an alphanumeric character.
+  // 2. Values can be any valid strings.
+  //
+  // Keys with empty prefix are reserved for end-users.
+  // Properties set by external tools or internally by the participant must use non-empty key prefixes.
+  //
+  // Duplicate keys are disallowed by the semantics of the protobuf3 maps.
+  // See: https://developers.google.com/protocol-buffers/docs/proto3#maps
+  //
+  // Annotations map may be a part of a modifiable resource whose update rpc call uses an update mask ('google/protobuf/field_mask.proto').
+  // For such update calls you may explicitly specify the update semantics of the 'annotations' map by appending one of the two modifiers to the update mask's path:
+  // (1) '!merge' to use merge semantics,
+  // (2) '!replace' to use replace-all semantics.
+  // Example path: 'user.metadata.annotations!replace'.
+  // NOTE: Merge semantics is used by default.
```

Comment:
TODO: 
Clarify that:
- 'merge' is the default semantics for non-default protobuf values,
- 'replace' is the default semantics for default protobuf values (because it's equivalent to a deletion),
- an update mask with an explicit '!merge' and a default value is an invalid request